.github/workflows/check_for_new_files.yml

```diff
@@ -0,0 +1,35 @@
+name: Validate CODEOWNERS
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  checkIfCodeownersUpdateNeeded:
+    permissions:
+      pull-requests: write
+    name: Check if CODEOWNERS needs to be updated
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Get count of added files
+        uses: actions/github-script@v6.4.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const { getAddedFiles } = require('./.github/workflows/scripts/check_for_new_files.js');
+            getAddedFiles({github, context, core});
```

Comment:
FYI, you can [return string data](https://github.com/actions/github-script#result-encoding) from `github-script` and reference it in later steps. 

I personally prefer that because it's more clear where the `NEW_FILES_COUNT` is coming from in later steps, but I'll leave the decision to you!